src/redstone_tx_sync/syncer.go

```diff
@@ -155,9 +166,68 @@ func (self *Syncer) checkTxForData(tx *types.Transaction, data string, ctx conte
 }
 
 func (self *Syncer) getTxSenderHash(tx *types.Transaction) (txSenderHash string, err error) {
-	signer := types.LatestSignerForChainID(tx.ChainId())
-	sender, err := types.Sender(signer, tx)
-	txSenderHash = sender.Hash().String()
+	sender, err := types.Sender(types.LatestSignerForChainID(tx.ChainId()), tx)
+	txSenderHash = sender.String()
+	return
+}
+
+func (self *Syncer) getSenderRoles(sender string) (roles *[]string, err error) {
+	senderDiscordIdPayload, err := self.getSenderDiscordId(sender)
+	if err != nil {
+		self.Log.WithError(err).Warn("Culd not retrieve sender id")
+		return
+	}
+
+	senderId := senderDiscordIdPayload.Result.UserId
+
+	resp2, err := self.httpClient.SetBaseURL(self.Config.RedstoneTxSyncer.SyncerWarpyApiUrl).R().
+		SetResult([]string{}).
+		SetQueryParams(map[string]string{
+			"id": senderId,
+		}).
+		SetHeader("Accept", "application/json").
+		Get("/v1/userRoles")
+
+	if err != nil {
+		self.Log.WithError(err).Warn("Culd not retrieve sender roles")
+		return
+	}
+
+	roles, ok := resp2.Result().(*[]string)
+	if !ok {
+		err = errors.New("Failed to parse response")
+		return
+	}
+	return
+}
+
+func (self *Syncer) getSenderDiscordId(sender string) (senderIdPayload *SenderDiscordIdPayload, err error) {
+	input, err := json.Marshal(struct {
+		Function string `json:"function"`
+		Address  string `json:"address"`
+	}{
+		Function: "getUserId",
+		Address:  sender,
+	})
+	resp, err := self.httpClient.SetBaseURL(self.Config.RedstoneTxSyncer.SyncerDreUrl).R().
+		SetResult(&SenderDiscordIdPayload{}).
```

Comment:
`		ForceContentType("application/json").` - this way it'll always try to parse json even if content type is missing.